File: training/csharp/src/De.Sots.CellarsAndCentaurs/Application/CreatureLookup.cs
using De.Sots.CellarsAndCentaurs.Domain.Model;
using De.Sots.CellarsAndCentaurs.Domain.Service;

namespace De.Sots.CellarsAndCentaurs.Application;

public class CreatureLookup
{
    private readonly CreatureService creatureService;

    public CreatureLookup(CreatureService creatureService)
    {
        this.creatureService = creatureService;
    }

    public Creature? FindOrNull(CreatureId id)
    {
        try
        {
            return creatureService.Find(id);
        }
        catch (NoSuchCreatureException)
        {
            return null;
        }
    }
}

File: sample-projects/php/src/Application/CreatureLookup.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Application;

use De\Sots\CellarsAndCentaurs\Domain\Model\Creature;
use De\Sots\CellarsAndCentaurs\Domain\Model\CreatureId;
use De\Sots\CellarsAndCentaurs\Domain\Model\NoSuchCreatureException;
use De\Sots\CellarsAndCentaurs\Domain\Service\CreatureService;

final class CreatureLookup
{
    private CreatureService $creatureService;

    public function __construct(CreatureService $creatureService)
    {
        $this->creatureService = $creatureService;
    }

    public function findOrNull(CreatureId $id): ?Creature
    {
        try {
            return $this->creatureService->find($id);
        } catch (NoSuchCreatureException $e) {
            return null;
        }
    }
}
